cx_Freeze/finder.py

```diff
@@ -412,77 +291,98 @@ def _InternalImportModule(self, name, deferredImports):
             if parentModule is None:
                 return None
             path = parentModule.path
-            searchName = name[pos + 1 :]
 
         if name in self.aliases:
             actualName = self.aliases[name]
             module = self._InternalImportModule(actualName, deferredImports)
             self._modules[name] = module
             return module
 
-        # detect namespace packages
-        try:
-            spec = importlib.util.find_spec(name)
-        except (AttributeError, ImportError, TypeError, ValueError):
-            spec = None
-        if (
-            spec
-            and spec.origin in (None, "namespace")
-            and spec.submodule_search_locations
-        ):
-            path = list(spec.submodule_search_locations)[0]
-            return self._LoadNamespacePackage(name, path, parentModule)
-        # other modules or packages
         try:
-            fp, path, info = self._FindModule(searchName, path)
-            if info[-1] == imp.C_BUILTIN and parentModule is not None:
-                return None
-            module = self._LoadModule(
-                name, fp, path, info, deferredImports, parentModule
+            module = self._load_module(
+                name, path, deferredImports, parentModule
             )
         except ImportError:
             logging.debug("Module [%s] cannot be imported", name)
             self._modules[name] = None
             return None
         return module
 
-    def _LoadModule(self, name, fp, path, info, deferredImports, parent=None):
-        """Load the module, given the information acquired by the finder."""
-        suffix, mode, type = info
-        if type == imp.PKG_DIRECTORY:
-            return self._LoadPackage(name, path, parent, deferredImports)
-        module = self._AddModule(name, file_name=path, parent=parent)
+    def _load_module(
+        self,
+        name: str,
+        path: Union[str, List[str]],
+        deferredImports: List[str],
+        parent: Optional["Module"] = None,
+    ) -> "Module":
```

Comment:
Shouldn't it be `Optional["Module"]`, because function can return `None` in few cases?

Reply:
Optional should be used for arguments.
https://docs.python.org/3/library/typing.html?highlight=typing#typing.Optional

Reply:
No, you can use Optional for return type (and other types). For example you can check this if you run mypy for this file (or use pycharm, it will give you warning about return type). https://mypy.readthedocs.io/en/stable/kinds_of_types.html#optional-types-and-the-none-type

Reply:
Actually, there places, where `Module` expected, but `Optional['Module']` returned, so here can be bugs.

Reply:
For what it's worth, I think the "Optional" might also be appropriate on the return type.  No affect on the code, but it will help the IDEs.

Reply:
@cainesi we talk exactly about return type
```python
) -> "Module":
```

Reply:
The question is: We can or we shall?
I understood that it was not necessary, from the documentation, that talks about an argument. I had also seen a response in the SO that made me continue my understanding.
As I am not a native English speaker, I may have misunderstood. I had understood arguments as parameters.
But I do not see the possibility of "bugs" because it is just a type annotation, and at runtime, it is ignored.
For us, the usefulness is that @cainesi well said.

Reply:
I took a look at the documentation for "Optional".

The reason it talks about arguments is to explain that "Optional" is *not* the same thing as a function argument with a default value. It looks like "Optional" can be used anywhere you are providing a type hint. 

If you have any english questions let me know. I'm up in Canada.

Reply:
> If you have any english questions let me know. I'm up in Canada.

@cainesi, In general, my English is good, and I use Grammarly to give me suggestions. Maybe for writing, my English is better than my Portuguese ;-)
Sometimes I can misunderstand some words that are not well explained, as I previously commented.
Thank you and @klensy for pointing me the mistake.
I hope this is the only problem with this code change.